<h2>Offline HBV-GLUE</h2>
<p>
The HBV-GLUE resource can be used "offline" to organise and analyse sequence data on a private computer.  
Offline HBV-GLUE takes the form of a GLUE project build (a linked dataset and set of analysis functions).
This project build can be loaded into an instance of the GLUE engine. 
</p>
<p>
A certain level of Unix command-line computing experience is required to install and use Offline HBV-GLUE. 
Please follow the instructions below to set up and use offline HBV-GLUE on your computer. A working installation
of offline HBV-GLUE is capable of a range of analysis functions. A couple are given below; the 
<a href="http://glue-tools.cvr.gla.ac.uk">GLUE engine website</a> documents how these functions may be accessed. 

<ol>
	<li><a ng-click="scrollTo('installGlue')">Install the GLUE engine</a></li>
	<li><a ng-click="scrollTo('downloadBuild')">Download an HBV-GLUE project build</a></li>
	<li><a ng-click="scrollTo('installBuild')">Install the project build into the GLUE database</a></li>
	<li><a ng-click="scrollTo('checkIntegrations')">Check that the MAFFT, RAxML and BLAST+ integrations are working correctly</a></li>
</ol>

<p>Please contact <a href="mailto:josh.singer@glasgow.ac.uk">Josh Singer</a> or post a question on the 
<a href="https://groups.google.com/forum/#!forum/glue-support">GLUE support forum</a> with any questions or 
queries about offline HBV-GLUE</p>

<h3 id="installGlue">Install the GLUE engine</h3>

Please follow the <a href="http://glue-tools.cvr.gla.ac.uk/#/installation">GLUE installation instructions</a>. 
The BLAST+, MAFFT and RAxML integration steps are required for most uses of Offline HBV-GLUE.

<h3 id="downloadBuild">Download an HBV-GLUE project build</h3>

<p>
There are two alternative builds for Offline HBV-GLUE. The smaller build includes the minimum set of packages for 
genotyping. The larger build also includes the NCBI-HBV-GLUE extension, i.e. HBV sequences 
derived from the NCBI nucleotide database and their metadata.
</p>	
<table class="table table-striped table-bordered table-nonfluid">
	<thead>
		<colgroup>
	       <col span="1" style="width: 50%;">
	       <col span="1" style="width: 50%;">
	    </colgroup>
	</thead>
	<tbody>
		<tr>
			<td><b>HBV-GLUE core project</b></td>
			<td>Download <a href="/hbv_glue_dbs/hbv_glue.sql.gz">hbv_glue.sql.gz</a></td>
		</tr>
		<tr>
			<td><b>HBV-GLUE core project + NCBI-HBV-GLUE extension</b></td>
			<td>Download <a href="/hbv_glue_dbs/ncbi_hbv_glue.sql.gz">ncbi_hbv_glue.sql.gz</a></td>
		</tr>
	</tbody>
</table>

<h3 id="installBuild">Install the project build into the GLUE database</h3>
<p>
<b>Warning:</b> This step will overwrite any data currently in the GLUE database.
</p>
<p>
Unzip the package and load into your MySQL databse using a Unix command line, something like this:
<div hljs language="bash" no-escape>
$ gunzip -c ncbi_hbv_glue.sql.gz | /usr/local/mysql/bin/mysql --user=gluetools --password=glue12345 GLUE_TOOLS
</div>
</p>
<p>Start the GLUE command line and use the <code>list project</code> GLUE command to validate that the project installed.</p>
<p>
<div hljs language="bash" no-escape>
<pre>
% gluetools.sh
GLUE version 1.1.66
Mode path: /
GLUE> list project
+======+============================================+
| name |                description                 |
+======+============================================+
| hbv  | A GLUE project for Hepatitis B virus (hbv) |
+======+============================================+
Projects found: 1</pre>
</div>
</p>

<h3 id="checkIntegrations">Check that the MAFFT, RAxML and BLAST+ integrations are working correctly</h3>
<p>
We will use the <code>hbvMaxLikelihoodGenotyper</code> module within HBV-GLUE to assign a genotype and
subgenotype for a sequence in a FASTA file. This will validate that MAFFT and RAxML are integrated and working correctly. 
</p>
<p>
Download the <a ng-click="downloadExampleSequence()">example sequence file</a> and make sure a copy
is present inside the directory from which you launched GLUE.</p>

<p>
Enter the GLUE commands indicated below and check the output.
<div hljs language="bash" no-escape>
<pre>
GLUE> project hbv 
OK
Mode path: /project/hbv
GLUE> module hbvMaxLikelihoodGenotyper genotype file --fileName fullGenome1.fasta
+=============+===================+====================+=======================+
| queryName   | speciesFinalClade | genotypeFinalClade | subgenotypeFinalClade |
+=============+===================+====================+=======================+
| fullGenome1 | AL_MASTER         | AL_A               | AL_A2                 |
+=============+===================+====================+=======================+</pre> 
</div>
</p>
<p>
We will use the <code>hbvFastaSequenceReporter</code> module within HBV-GLUE to translate the RT region of this sequence to amino acids. 
This step uses BLAST+ to align the sequence with a reference, and so if the step works it validates that BLAST+ is integrated and working correctly.
 
<p>
Enter the GLUE command indicated below and check the output. Use 'Q' to exit the interactive table.
<div hljs language="bash" no-escape>
<pre>
GLUE> module hbvFastaSequenceReporter amino-acid -i fullGenome1.fasta -r REF_NUMBERING_X02763 -f RT -t REF_A2_EU594385 -a AL_UNCONSTRAINED
+=========+============+=========+==========+============+==========+===========+============+============+
| feature | codonLabel | queryNt | relRefNt |dependentNts| codonNts | aminoAcid |definiteAas |possibleAas |
+=========+============+=========+==========+============+==========+===========+============+============+
| RT      | 1          | 130     | 130      |GAG         | GAG      | E         |E           |E           |
| RT      | 2          | 133     | 133      |GAC         | GAC      | D         |D           |D           |
| RT      | 3          | 136     | 136      |TGG         | TGG      | W         |W           |W           |
| RT      | 4          | 139     | 139      |GGA         | GGA      | G         |G           |G           |
| RT      | 5          | 142     | 142      |CCC         | CCC      | P         |P           |P           |
| RT      | 6          | 145     | 145      |TGT         | TGT      | C         |C           |C           |
| RT      | 7          | 148     | 148      |GAC         | GAC      | D         |D           |D           |
| RT      | 8          | 151     | 151      |GAA         | GAA      | E         |E           |E           |
| RT      | 9          | 154     | 154      |CAT         | CAT      | H         |H           |H           |
| RT      | 10         | 157     | 157      |GGA         | GGA      | G         |G           |G           |
| RT      | 11         | 160     | 160      |GAA         | GAA      | E         |E           |E           |
| RT      | 12         | 163     | 163      |CAT         | CAT      | H         |H           |H           |
| RT      | 13         | 166     | 166      |CAC         | CAC      | H         |H           |H           |
| RT      | 14         | 169     | 169      |ATC         | ATC      | I         |I           |I           |
| RT      | 15         | 172     | 172      |AGG         | AGG      | R         |R           |R           |
| RT      | 16         | 175     | 175      |ATT         | ATT      | I         |I           |I           |
| RT      | 17         | 178     | 178      |CCT         | CCT      | P         |P           |P           |
| RT      | 18         | 181     | 181      |AGG         | AGG      | R         |R           |R           |
| RT      | 19         | 184     | 184      |ACC         | ACC      | T         |T           |T           |
| RT      | 20         | 187     | 187      |CCT         | CCT      | P         |P           |P           |
+=========+============+=========+==========+============+==========+===========+============+============+
Rows 1 to 20 of 344 [F:first, L:last, P:prev, N:next, Q:quit]
</div>
</p>
 